sound/soc/intel/boards/sof_es8336.c

```diff
@@ -469,24 +469,16 @@ static int sof_es8336_probe(struct platform_device *pdev)
 
 	card = &sof_es8336_card;
 	card->dev = dev;
-
-	if (!dmi_check_system(sof_es8336_quirk_table)) {
+	
+	if (mach->mach_params.i2s_link_mask & BIT(2))
 		quirk = SOF_ES8336_SSP_CODEC(2);
-	} else {
-		/*
-		 *  Set configuration based on platform NHLT.
-		 * In this machine driver, we can only support one SSP for the
-		 * ES8336 link, the else-if below are intentional
-		 */
-		if (mach->mach_params.i2s_link_mask & BIT(0))
-			quirk = SOF_ES8336_SSP_CODEC(0);
-		else if (mach->mach_params.i2s_link_mask & BIT(1))
-			quirk = SOF_ES8336_SSP_CODEC(1);
-		else if (mach->mach_params.i2s_link_mask & BIT(2))
-			quirk = SOF_ES8336_SSP_CODEC(2);
-		if (mach->mach_params.dmic_num)
-			quirk |= SOF_ES8336_ENABLE_DMIC;
-	}
+	else if (mach->mach_params.i2s_link_mask & BIT(1))
+		quirk = SOF_ES8336_SSP_CODEC(1);
+	else
+		quirk = SOF_ES8336_SSP_CODEC(0);
+	if (mach->mach_params.dmic_num)
+		quirk |= SOF_ES8336_ENABLE_DMIC;
+
```

Comment:
@yangxiaohua2009 I don't understand this patch. 
The idea was that if we find a DMI match we use the information as is, otherwise we use what is provided by NHLT. It happen that some BIOSes are completely wrong and we cannot trust the information blindly.

What you are suggesting here seems to mean that on some platform the DMI matching is incorrect and selects SSP2 instead of SSP0. That should be a separate issue, no?

Can you clarify on which platform you are testing and how come the SSP2 is used, is this really because the DMI match fails?

Reply:
never mind, got your point and will fix this.